src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -266,24 +269,15 @@ await listener.AfterExecutionAsync(context)
                         }
                 }
 
-                if (context.Errors.Count > 0)
-                {
-                    result.Errors = context.Errors;
-                }
+                result.AddErrors(context.Errors);
```

Comment:
Why are we making these changes?  It just adds more allocations during execution.  Plus the code is harder to read.

Reply:
It's fine though.  Optimizing runtime when errors are present isn't a priority.

Reply:
Why more ?

Reply:
My thought was that errors from the context should complement the errors already recorded in the result, and not replace them. This is not true?

Reply:
What if `result.Errors` is not null/empty?

Reply:
With `AddErrors`, it has to iterate over the errors in `context.Errors` and add them to a new instance within `result`.  So it is more allocations.

Reply:
> My thought was that errors from the context should complement the errors already recorded in the result, and not replace them. This is not true?

Makes sense, although there is no existing possibility of this occurring, unless perhaps a completely custom executer was used.

Reply:
> It's fine though. Optimizing runtime when errors are present isn't a priority.

I have not optimized.

> With AddErrors, it has to iterate over the errors in context.Errors and add them to a new instance within result. So it is more allocations.

OK. Possible one `List` instance.

> Makes sense, although there is no existing possibility of this occurring, unless perhaps a completely custom executer was used.

I just didn't like that I saw the potential for losing errors. I made sure that they are not lost in any situation, without understanding whether it is possible now or in the future. Is it good?

Reply:
👍